### PR TITLE
Redirect helper

### PR DESCRIPTION
**Pull request by ReneSansSerif**
Introduce helper function "redirect()":
It works like header() but you can optionally send data with it (for example $_POST) in order to restore form field values.
Data can be retrieved in the views via associative array $_SESSION["old_POST"] .
It is important to unset this array at the end of every view file.

### CODE REVIEW

**Comment by eXtgmA** [comment]
# PHPSTAN Result
```::error file=src/helper/redirect.php,line=14,col=0::Function redirect() has parameter $old_POST with no value type specified in iterable type array.
 ------ -------------------------------------------------------------------------------------- 
  Line   src/helper/redirect.php                                                               
 ------ -------------------------------------------------------------------------------------- 
  14     Function redirect() has parameter $old_POST with no value type                        
         specified in iterable type array.                                                     
         💡 See:                                                                               
            https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type  
 ------ -------------------------------------------------------------------------------------- 


 [ERROR] Found 1 error                                                          
```

**Comment by eXtgmA** [comment]
# Autofix Result
Fix code style issues automatically
```
PHPCBF RESULT SUMMARY
----------------------------------------------------------------------
FILE                                                  FIXED  REMAINING
----------------------------------------------------------------------
...cation_rental/vacation_rental/src/models/User.php  5      0
...acation_rental/src/views/partials/footer.view.php  4      0
...on_rental/vacation_rental/src/helper/redirect.php  3      0
----------------------------------------------------------------------
A TOTAL OF 12 ERRORS WERE FIXED IN 3 FILES
----------------------------------------------------------------------

Time: 696ms; Memory: 8MB

```

File: src/controller/BaseController.php [hunk 1]
```diff
@@ -1,6 +1,8 @@
 <?php
 namespace src\controller;
 
+include_once("../src/helper/redirect.php");
```
Comment (BBuermann-IT) [inline src/controller/BaseController.php:4]: nice

File: src/helper/redirect.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ *
+ * @param string $headerstring
+ * @param int $response_code
+ * @param array|null $old_POST
+ * @param bool $replace
+ * @return void
+ */
+function redirect(string $headerstring, int $response_code, array $old_POST = null, bool $replace = true): void
+{
+
```
Comment (BBuermann-IT) [inline src/helper/redirect.php:16]: looks good

**Review by BBuermann-IT — APPROVED** [review]
Sieht gut aus

**Pull request merged**